plugin-build/plugin/build.gradle.kts

```diff
@@ -0,0 +1,66 @@
+import com.vanniktech.maven.publish.MavenPublishPluginExtension
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    kotlin("jvm")
+    `java-gradle-plugin`
+    alias(libs.plugins.vanniktech.publish)
+    id("distribution")
+}
+
+dependencies {
+    implementation(kotlin("stdlib"))
```

Comment:
`m`: I think these all should be `compileOnly` otherwise we'd be bundling kgp and gradle-api with our plugin and potentially overwriting what users use in their build